src/MvvmNano.Forms/MvvmNanoApplication.cs

```diff
@@ -7,6 +8,26 @@ namespace MvvmNano.Forms
     /// </summary>
     public class MvvmNanoApplication : Application
     {
+
+        /// <summary>
+        /// A collection of MasterDetailData that represents available detail pages.
+        /// </summary>
+        public ObservableCollection<MasterDetailData> MasterDetails { get; set; } = new ObservableCollection<MasterDetailData>();
```

Comment:
Maybe a MvvmNanoMasterDetailApplication that overrides the SetMainPage() instead of introducing SetMasterDetailPage() and contains the "master detail stuff"